File: codility/rotating_array.ts
// Rotate an array to the right by a given number of steps.
function solutionTest(A: number[], K: number): number[] {
    if (A.length === 0){
        return A;
    }
    if (K === 0){
        return A;
    }
    K = K % A.length;
    const part1 = A.slice(0, A.length - K);
    const part2 = A.slice(A.length - K);
    return part2.concat(part1);

}

const A = [1, 2, 3, 4];
const K = 4;
console.log(solutionTest(A, K));

File: codility/python/rotating_array.py
#Rotate an array to the right by a given number of steps.
def solution(A, K):
    # Implement your solution here
    #check if array A  is empty, K too
    # normalize the number of rotations
    # split into two parts and
    if len(A) == 0:
        return A
    if K == 0:
        return A
    K = K % len(A)
    part_one = A[-K:]
    part_two = A[:-K]

    rotated_array = part_one + part_two
    return rotated_array
